src/main/java/com/epam/aidial/core/security/ApiKeyStore.java

```diff
@@ -2,57 +2,117 @@
 
 import com.epam.aidial.core.config.ApiKeyData;
 import com.epam.aidial.core.config.Key;
+import com.epam.aidial.core.data.ResourceType;
+import com.epam.aidial.core.service.LockService;
+import com.epam.aidial.core.service.ResourceService;
+import com.epam.aidial.core.storage.ResourceDescription;
+import com.epam.aidial.core.util.ProxyUtil;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import lombok.AllArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 
 import java.util.HashMap;
-import java.util.Iterator;
 import java.util.Map;
+import javax.annotation.concurrent.GuardedBy;
 
 import static com.epam.aidial.core.security.ApiKeyGenerator.generateKey;
+import static com.epam.aidial.core.storage.BlobStorageUtil.PATH_SEPARATOR;
 
 @Slf4j
+@AllArgsConstructor
```

Comment:
Can you add javadocs for this class or method with some hints how to use it?
For example ApiKeyData stored locally in `keys` and related per-request-keys stored in redis only

Reply:
added